### ISSUE TITLE
Pure procedures

### ISSUE THREAD

**Issue by gokhanyalniz** [burakbudanur/dnsbox#1]
https://en.wikibooks.org/wiki/Fortran/Fortran_procedures_and_functions#Pure_procedures
There is the syntax `pure` to denote functions and subroutines that have no side effects. Apparently compilers can optimize code based on that, but I find it valuable more as a sanity check. This fits nicely to our latest code cleanup: almost all procudures are already pure, the exceptions are almost all `MPI_REDUCE` related, and the (now) rare, purpose specific global arrays or scalars (projections, stats, phase angles etc.) I think it would nice to have this annotation around.

 - Procedures that are called only once need not be pure, like various inits.
